### PR TITLE
Added prefix and delimiter support to file list API call.

### PR DESCRIPTION
**Pull request by crazyscience**

### CODE REVIEW

**Comment by yakovkhalinsky** [comment]
Thanks @crazyscience for the PR.

Are you able to update the failing tests and push another commit.

Then I can merge and publish 👍

**Comment by crazyscience** [comment]
@yakovkhalinsky There ya go.

File: lib/actions/file.js [hunk 1]
```diff
@@ -41,7 +44,10 @@ exports.listFileNames = function(b2, args) {
         body: JSON.stringify({
             bucketId: bucketId,
             startFileName: startFileName ? startFileName : '',
-            maxFileCount: maxFileCount ? maxFileCount : 100
+            maxFileCount: maxFileCount ? maxFileCount : 100,
+            prefix: prefix ? prefix : '',
+            delimiter: delimiter ? delimiter : null
```
Comment (yakovkhalinsky) [inline lib/actions/file.js]: Just curious, why is `delimiter` defaulted to `null` when the `prefix` is defaulted to `blank`?

Comment (crazyscience) [inline lib/actions/file.js]: @yakovkhalinsky Those are the default values that the B2 API expects (see documentation). We were working on a project today that includes/needs these modifications and ran into an issue with `delimiter = ''`. The API response was pretty clear that, if included, `delimiter` must be a string with a length of 1. The same rules don't apply with `prefix` as it defaults to an empty string anyway.

Comment (yakovkhalinsky) [inline lib/actions/file.js]: Cool fair enough 👍 

Merging

Comment (crazyscience) [inline lib/actions/file.js]: :+1:

Comment (crazyscience) [inline lib/actions/file.js]: Another one on the way for you in the next hour or so.

Comment (yakovkhalinsky) [inline lib/actions/file.js]: Cheer @crazyscience I'll hold off publishing for the next PR 👍 

That PR and this one will go out as `0.9.12`

**Pull request merged**